### PR TITLE
Interactive markers

### PR DESCRIPTION
**Pull request by lornaarmstrong**
when a marker (not hardcoded) is clicked...
- fetch recommendation with ID matching that of the marker
- create Recommendation object
- update HTML to show that information

TODO: Fix issues. Currently, the listener events are causing a problem - all markers show the same information from one recommendation in the datastore, rather than the corresponding.

### CODE REVIEW

**Comment by lornaarmstrong** [comment]
NOTE: 
Commits already reviewed (from other branch), except: Fetching recommendation information from datastore (bc9d426)

**Comment by lornaarmstrong** [comment]
NOTE: The problem (regarding the TODO above) was being caused because there was no keyword to declare the marker. Thanks to @MostafaGabriel for spotting it!

File: src/main/java/com/google/sps/servlets/RecommendationServlet.java [hunk 1]
```diff
@@ -56,11 +83,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
     int costRating = Integer.parseInt(request.getParameter("price"));
     int crowdRating = Integer.parseInt(request.getParameter("crowd"));
-    // Create entity on recommendation data
+
+    // Create entity from recommendation data
     Entity recommendationEntity = new Entity("Recommendation");
     recommendationEntity.setProperty("place-name", name);
     recommendationEntity.setProperty("latitude", lat);
-    recommendationEntity.setProperty("longitude", lng);
+    recommendationEntity.setProperty("longitude", lng);
```
Comment (raniarusly) [inline src/main/java/com/google/sps/servlets/RecommendationServlet.java]:
```suggestion
    recommendationEntity.setProperty("longitude", lng);
```

File: src/main/webapp/scripts/map.js [hunk 2]
```diff
@@ -196,12 +196,50 @@ function placeMarkerAndPanTo(latLng, map) {
...
+  /* Add Listener for Click on Marker */
+   marker.addListener("click", () => {
+     fetchRecommendationInfo(marker.id);
+   });
+}
+
+/* Get recommendation from datastore and fill html */
+function fetchRecommendationInfo(id) {
+  const params = new URLSearchParams();
+  params.append('id', id);
```
Comment (raniarusly) [inline src/main/webapp/scripts/map.js]: I think params is not used anywhere 
```suggestion
```

File: src/main/webapp/scripts/map.js [hunk 3]
```diff
@@ -196,12 +196,50 @@ function placeMarkerAndPanTo(latLng, map) {
...
+  fetch("/recommendation?id=" + id).then(result => result.json()).then((recommendation) => {
+    /* Update the HTML */
+      document.getElementById("category-header").innerHTML = formatCategory(recommendation.category);
+      document.getElementById("category-header").style.backgroundColor = getBackgroundColour(
+          formatCategory(recommendation.category)
+        );
+      document.getElementById("place-title").innerHTML = recommendation.name;
+      document.getElementById("rec-address").innerHTML = recommendation.lat+", "+recommendation.lng;
+      document.getElementById("place-recommendation").innerHTML = recommendation.description;
+      document.getElementById("rec-container").style.display = "block";
```
Comment (raniarusly) [inline src/main/webapp/scripts/map.js]: nit
```suggestion
    const formattedCategory = formatCategory(recommendation.category);
    document.getElementById("category-header").innerHTML = formatCategory(formattedCategory);
    document.getElementById("category-header").style.backgroundColor = getBackgroundColour(formattedCategory);
    document.getElementById("place-title").innerHTML = recommendation.name;
    document.getElementById("rec-address").innerHTML = recommendation.lat + ", " + recommendation.lng;
    document.getElementById("place-recommendation").innerHTML = recommendation.description;
    document.getElementById("rec-container").style.display = "block";
```

File: src/main/webapp/scripts/map.js [hunk 4]
```diff
@@ -196,12 +196,50 @@ function placeMarkerAndPanTo(latLng, map) {
...
+  });
+}
+
+/* Formats the category string */
+function formatCategory(category) {
+  switch(category) {
+    case 'restaurants':
```
Comment (raniarusly) [inline src/main/webapp/scripts/map.js]: nit: should we use double quote just so that it is all uniform?

**Review by raniarusly — APPROVED** [review]
only nits, thank you! :D

File: src/main/java/com/google/sps/servlets/RecommendationServlet.java [hunk 5]
```diff
@@ -35,7 +41,28 @@ public void init() {
...
+      Key key = KeyFactory.createKey("Recommendation", id);
+      Entity recommendation = datastore.get(key);
+      // Get details of recommendation
+      String name = (String) recommendation.getProperty("place-name");
+      String category = (String) recommendation.getProperty("category");
+      double lat = (double) recommendation.getProperty("latitude");
+      double lng = (double) recommendation.getProperty("longitude");
+      String description = (String) recommendation.getProperty("description");
+      int costRating = (int) (long) recommendation.getProperty("cost-rating");
+      int crowdRating = (int) (long) recommendation.getProperty("crowd-rating");
```
Comment (MostafaGabriel) [inline src/main/java/com/google/sps/servlets/RecommendationServlet.java:58]: Would be nice if you please clarify here why we are casting twice :) [Example](https://github.com/googleinterns/step241-2020/blob/master/src/main/java/com/google/sps/servlets/UserDataServlet.java#L50-L51)

File: src/main/webapp/scripts/map.js (hunk 2 above)
Comment (MostafaGabriel) [reply src/main/webapp/scripts/map.js]: Good spot! Please either remove the unused variable or use it instead of appending the `id` value in fetch

**Review by MostafaGabriel — APPROVED** [review]
LGTM! Thanks Lorna :)

File: src/main/java/com/google/sps/servlets/RecommendationServlet.java (hunk 5 above)
Comment (lornaarmstrong) [reply src/main/java/com/google/sps/servlets/RecommendationServlet.java:58]: Good point, thank you!

**Pull request merged**